src/aggregate/aggregate_request.c

```diff
@@ -1300,6 +1300,15 @@ static void PushUpStream(ResultProcessor *rp_to_place, ResultProcessor *rp) {
   rp_to_place->parent = rp->parent;
 }
 
+// We need to know if the last accumulator is wrapped by a result processor that accesses redis, and therefore
+// is wrapped by a buffer locker and an unlocker. If so, and we are in cursor mode, we need to push the buffer locker
+// downstream after the first cursor read (before the first `FT.CURSOR READ` command). We therefore turn on the flag
+// QEXEC_S_NEEDS_BUFFER_REEVAL in the query execution state.
+// To follow the pipeline and find out whether we wrap the last accumulator or not, we use the following state machine:
+typedef enum { WRAPPING, NOT_WRAPPING, UNSET, SEEN_ACCUMULATOR_BEFORE_UNLOCKER } WrapsLastAccumulatorState;
+// the `WRAPPING` and `NOT_WRAPPING` states are terminal states, and the `UNSET` state is the initial state.
```

Comment:
```suggestion
// the `WRAPPING` and `NOT_WRAPPING` states are terminal states, and the `SEEN_ACCUMULATOR_BEFORE_UNLOCKER` state is the initial state.
```

Reply:
`UNSET` is the initial state. `SEEN_ACCUMULATOR_BEFORE_UNLOCKER` is a state that you must go through to reach `WRAPPING`, but it's not the initial one